lib/Conch/Controller/Login.pm

```diff
@@ -95,35 +100,43 @@ sub authenticate ($c) {
 		my $user = $c->db_user_accounts->lookup_by_name($name);
 
 		unless ($user) {
-			$c->status( 401, { error => 'unauthorized' } );
-			return 0;
+			$c->log->debug('basic auth failed: user not found');
+			return $c->status(401, { error => 'unauthorized' });
 		}
 
-		if ($user->validate_password($password)) {
-			$c->stash( user_id => $user->id );
-			$c->stash( user    => $user );
-			return 1;
+		if (not $user->validate_password($password)) {
+			$c->log->debug('basic auth failed: incorrect password');
+			return $c->status(401, { error => 'unauthorized' });
 		}
-		else {
-			$c->status( 401, { error => 'unauthorized' } );
-			return 0;
+
+		if ($user->force_password_change) {
+			$c->log->debug('basic auth failed: password correct, but force_password_change was set');
+			return $c->status(401, {
+				error => 'unauthorized',
+				message => 'reset password at /user/me/password within 10 minutes',
```

Comment:
I'd like this converted to something other than a text string, particularly one in a 'message' element. Clients would have to parse this message to figure out if they need to be concerned.

Reply:
Add a Location header.